### ISSUE TITLE
Bug: Validation rule can't handle array data

### ISSUE THREAD

**Issue by shishamo** [codeigniter4/CodeIgniter4#5368]
PHP Version

7.4

CodeIgniter4 Version

4,1,5

CodeIgniter4 Installation Method

Composer (new project)

Which operating systems have you tested for this bug?

Linux

Which server did you use?

apache

Database

MariaDB 10.1.31

What happened?
I am trying to create a new rule to check if value "is array" and "array length equals to".
The parameter passed inside the new function doesn't get the array but the array first parameter.

Steps to Reproduce
New rule created as below
> is_array($value) {
 return is_array($value);
}

If you add the new rule inside the validator as below
> $data = [
 'arr' => [1,2,3,4,5],
];
$validation->setRules([
 'arr' => 'is_array',
]);

Validator always return false because the value passed inside the validator is 1 and not the entire array.

Expected Output
Need to get the entire array inside the validator?

Anything else?

**Comment by kenjis** [comment]
See https://codeigniter.com/user_guide/libraries/validation.html#creating-custom-rules

The first parameter is `string`. It seems you are doing wrong.

```php
class MyRules
{
    public function even(string $str): bool
    {
        return (int) $str % 2 == 0;
    }
}
```

Please ask your support questions in the forums. Thanks!
https://forum.codeigniter.com/forum-30.html

**Comment by shishamo** [comment]
@kenjis 
Sorry, maybe i didn't explain well...

Here is my rule:
```
class MyRules
{
    public function is_array(string $value): bool
    {
        d($value);
        if (is_array($value)) {
            return true;
        }
        return false;
    }
}
```

When i run the validation as below
```
$data['arr'] = [1,2,3,4,5];
$validation->setRules([
    'arr' => 'is_array',
]);
```

And dump the value passed inside the rule i created
Only the first value of the array is passed
```
int(1)
```

**Comment by kenjis** [comment]
I think the custom rule can get only `string` data, so there is no way to get an array.
So what you want to do is impossible with the current API.
Therefore I must say it is by design, not a bug.

If you find a way to handle an array, let me know.

**Comment by kenjis** [comment]
This  also may be a bug.

**Comment by kenjis** [comment]
@shishamo This bug was fixed.
It would be included in the next release v4.1.6.

**Comment by shishamo** [comment]
Thanks a lot @kenjis to you and your team~
